pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -325,18 +329,10 @@ func (builder *Dockerfile) Prepare(config *api.Config) error {
 		}
 	}
 
-	scriptInstaller := scripts.NewInstaller(
-		"",
-		config.ScriptsURL,
-		config.ScriptDownloadProxyConfig,
-		nil,
-		api.AuthConfig{},
-		builder.fs,
-	)
-
-	// all scripts are optional, we assume/hope the image contains scripts if we don't find them
-	// in the source repo.
-	scriptInstaller.InstallOptional(append(scripts.RequiredScripts, scripts.OptionalScripts...), config.WorkingDir)
+	// Install scripts provided by user
+	builder.installScripts(config.ScriptsURL, config)
+	// Install scripts specified by image metadata
+	builder.installScripts(config.ImageScriptsURL, config)
```

Comment:
assuming this does what i think it does, the order seems backwards.  We want "scriptsurl" to take precedence, so you need to install those scripts second, so they overwrite anything you got from "imagescriptsurl"